### PR TITLE
spread-objects-operation

### PR DESCRIPTION
**Pull request by pazhersh**
Added to the `{{}}` object template an option to spread an object value into it's parent.

### CODE REVIEW

File: lib/template.js [hunk 1]
```diff
@@ -93,6 +93,19 @@ const renderRecursively = (inputJson, template, execOptions = {}) => {
   if (typeof template === 'object' && template !== null) {
     return Object.fromEntries(
       Object.entries(template).flatMap(([key, value]) => {
+        const MY_KEYWORD_ESCAPED = "tempSpreadKeyword\\(\\)";
+        const keywordMatcher = `^\\{\\{\\s*${MY_KEYWORD_ESCAPED}\\s*\\}\\}$`;
+
+        if (key.match(keywordMatcher)) {
```
Comment (talsabagport) [inline lib/template.js]: maybe add trim to the key as it can starts/ends with whitespaces as well?

Comment (pazhersh) [reply lib/template.js]: The matcher has `\s`s to match whitespaces
There's also a test for whitespace trims 😉

File: test/template.test.js [hunk 2]
```diff
@@ -124,6 +124,8 @@ describe('template', () => {
         expect(render({'{{null}}': 'bar'})).toEqual({});
         expect(render({'{{""}}': 'bar'})).toEqual({});
         expect(render({'{{\'\'}}': 'bar'})).toEqual({});
+        expect(render({ "{{spreadValue()}}": { foo: "bar" } })).toEqual({foo: "bar"});
+        expect(render({ " {{ spreadValue() }} ": { foo: "bar" } })).toEqual({foo: "bar"});
```
Comment (talsabagport) [inline test/template.test.js]: Can you add a test for object value with string jq?

Comment (pazhersh) [reply test/template.test.js]: 👌 added some

**Review by talsabagport — APPROVED** [review]
